dataset/src/test/scala/frameless/TypedDatasetSuite.scala

```diff
@@ -6,11 +6,7 @@ import org.scalactic.anyvals.PosZInt
 import org.scalatest.FunSuite
 import org.scalatest.prop.Checkers
 
```

Comment:
I think the motivation is that `TypedDataset` (like vanilla `Dataset`) is lazy in some things but not others. Whenever you have an operation that stays in `TypedDataset` (like `map` or `select` or `flatMap`) it is lazy. But when you have an operation that moves out of `TypedDataset`, it is immediate - like `count` or `collect`. I guess the solution to that was to make these operations instead be natural transformations into another lazy TC which is `Job`, and it controls evaluation explicitly through `run`.

You could argue that the programmer just needs to distinguish between lazy and strict operations on `TypedDataset`, and that would be more in line with what Spark programmers already do. But I think there's value in making all operations lazy.

I share your trepidation that we have some operations that happen in one TC and other operations happen in another, and the only division is which things Spark happens to do strictly vs. lazily. But the only other alternative (besides ignoring the lazy vs strict distinction and just doing what Spark does) would be to have one outer monad transformer, and `TypedDataset` becomes that transformer over `Dataset`. I think that would be a whole lot more complicated than what we have now, even with `Job` being a weird outsider.

Like I mentioned elsewhere, if we forced a cats dependency then `Job` could be replaced with `Eval` and it might be a little less weird.